### ISSUE TITLE
Spurious CA2012 on lhs of pattern match

### ISSUE THREAD

**Issue by poizan42** [dotnet/roslyn-analyzers#7159]
### Analyzer

**Diagnostic ID**: [CA2012](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2012): `Use ValueTasks correctly`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.101](https://dotnet.microsoft.com/download/dotnet/8.0)

### Describe the bug

The following generates CA2012

```CSharp
if (x?.SomeActionReturningValueTaskAsync()) is ValueTask vt)
	await vt;
```

### Steps To Reproduce

Put the following code in any .cs file in Visual Studio

```CSharp
public class Foo
{
	public ValueTask FrobAsync() => default;
	public static async ValueTask FrobFooAsync(Foo? foo)
	{
		if (foo?.FrobAsync() is ValueTask vt)
			await vt;
	}
}
```

### Expected behavior

CA2012 should not be emitted as long as the ValueTask is only consumed once and not otherwise escapes.

### Actual behavior

CA2012 is emitted at the "." of `foo?.FrobAsync()`

**Comment by Cyberboss** [comment]
I suspect this is similar to this notice that plagues my codebase wherever I use my custom WhenAll for `ValueTask`s. (It accepts a `params ValueTask[]`)

![image](https://github.com/user-attachments/assets/7f10d152-5830-4370-bc66-022748e93219)

```cs
public ValueTask DisposeAsync()
	=> ValueTaskExtensions.WhenAll(
		RestClient.DisposeAsync(),
		GraphQLClient.DisposeAsync());
```
